### DISCUSSION TITLE
Unable to use STOMP with NATS(Java spring)

### DISCUSSION THREAD

**Discussion by roie16** [nats-io/nats.java#1196] [Q&A]
Hi all,
I have opened this [stackoverflow](https://stackoverflow.com/questions/78856083/spring-websocket-with-nats-as-stomp-broker-no-enum-constant-error/78856139#78856139) post,
Basically I can see that NATS is opening with WS port as it should, but I am unable to connect it as broker to the STOMP protocol and I am not sure why,
Can you please help? 
Roie

**Comment by scottf** [comment]
@roie16 I'm a little confused by this post. The NATS client knows nothing about the STOMP protocol. It talks to a NATS server with NATS protocol.

**Reply by roie16** [reply]
@scottf , I want to leverage NATS as the inbound outbound message broker In my app, because NATS support of websocket I assumed it was also supporting STOMP protocol(like RabbitMQ can use [STOMP over Websocket](https://www.rabbitmq.com/docs/web-stomp)), but I am starting to think that is not the case, I am thinking now that I may need to request NATS to add STOMP support
